airflow/www/views.py

```diff
@@ -2847,8 +2847,8 @@ def duration(self, dag_id, session=None):
             session.query(TaskFail)
             .filter(
                 TaskFail.dag_id == dag.dag_id,
-                TaskFail.execution_date >= min_date,
-                TaskFail.execution_date <= base_date,
+                TaskInstance.execution_date >= min_date,
+                TaskInstance.execution_date <= base_date,
```

Comment:
I wonder if we could directly use `DagRun.execution_date` here.

Reply:
could try it

Reply:
appears it does work.  that's pretty magical!